src/time.rs

```diff
@@ -0,0 +1,149 @@
+use sys::time::TimeSpec;
+use std::mem::MaybeUninit;
+use {Result, Errno};
+use libc;
+
+libc_enum! {
+    #[cfg_attr(any(
+                target_os = "fuchsia",
+                target_os = "redox",
+                target_env = "uclibc",
+                target_os = "freebsd",
+                target_os = "netbsd",
+                target_os = "openbsd",
+                target_os = "haiku",
+                target_os = "linux",
```

Comment:
This should be `all(target_os = "linux", not(target_env = "newlib"))` and likewise for android and emscripten.

Reply:
Ah, that's very tricky. Added that in the case of everything that appears after newlib/uclibc. In some cases it had to be `all(target_os = "<x>", not(target_env = "newlib", target_env = "uclibc"))`.